File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/about/AboutPresenter.java
package com.androidacademy.msk.exerciseproject.screen.about;

import com.androidacademy.msk.exerciseproject.screen.base.BasePresenter;
import com.arellomobile.mvp.InjectViewState;

@InjectViewState
public class AboutPresenter extends BasePresenter<AboutView> {

    public void onSendMessageButtonClicked() {
        getViewState().openSmsApp();
    }

    public void onEmailButtonClicked() {
        getViewState().openEmailApp();
    }

    public void onTelegramButtonClicked() {
        getViewState().openTelegramApp();
    }

    public void onInstagramButtonClicked() {
        getViewState().openIstagramApp();
    }
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/about/AboutPresenter.kt
package com.androidacademy.msk.exerciseproject.screen.about

import com.androidacademy.msk.exerciseproject.screen.base.BasePresenter
import com.arellomobile.mvp.InjectViewState

@InjectViewState
class AboutPresenter : BasePresenter<AboutView>() {

    fun onSendMessageButtonClicked() {
        viewState.openSmsApp()
    }

    fun onEmailButtonClicked() {
        viewState.openEmailApp()
    }

    fun onTelegramButtonClicked() {
        viewState.openTelegramApp()
    }

    fun onInstagramButtonClicked() {
        viewState.openIstagramApp()
    }
}
